test/Microsoft.AspNetCore.Razor.Runtime.Test/Runtime/TagHelpers/TagHelperExecutionContextTest.cs

```diff
@@ -289,7 +289,7 @@ public void HtmlAttributes_IgnoresCase(string originalName, string updatedName)
         {
             // Arrange
             var executionContext = new TagHelperExecutionContext("p", TagMode.StartTagAndEndTag);
-            executionContext.HTMLAttributes[originalName] = "hello";
+            executionContext.AddHtmlAttribute(originalName, "hello");
```

Comment:
As mentioned in the other PR, collection properties that can return `null` are the worst.